File: LeetCode/Solution_0268.java
// 268. Missing Number

class Solution {
    public int missingNumber(int[] nums) {
        int length = nums.length,curr = 0;
        int sum = (length*(length+1))/2;
        for(int i=0;i<length;i++){
            curr+=nums[i];
        }
        return sum-curr;
    }
}

// Time Complexity: O(n)
// Space Complexity: O(1)

File: LeetCode/Solution_0268.py
# 268. Missing Number

from typing import List

class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        n = len(nums)
        total = (n*(n+1))//2
        s = sum(nums)
        return total-s
    
# Time Complexity: O(n) - We traverse the list of numbers once
# Space Complexity: O(1) - We use a constant amount of space